python/paddle/device/cuda/__init__.py

```diff
@@ -141,3 +141,37 @@ def empty_cache():
 
     if core.is_compiled_with_cuda():
         core.cuda_empty_cache()
+
+
+def get_device_properties(device):
+    '''
+    Return the properties of given CUDA device.
+
+    Args:
+        device(paddle.CUDAPlace() or int): The device or the ID of the device which want to get the properties of the device from. 
+
+    Returns:
+        _CudaDeviceProperties: the properties of the device which include ASCII string identifying device, major compute capability, minor compute capability, 
+                               global memory available on device in bytes and the number of multiprocessors on the device.
+
+    Examples:
+    
+        .. code-block:: python
+
+            # required: gpu
+            import paddle
+
+            paddle.device.cuda.get_device_properties(0)
+            paddle.device.cuda.get_device_properties(paddle.CUDAPlace(0))
+
+    '''
+    device_id = -1
+    if device is not None:
+        if isinstance(device, int):
+            device_id = device
+        elif isinstance(device, core.CUDAPlace):
+            device_id = device.get_device_id()
+        else:
+            raise ValueError("device type must be int or paddle.CUDAPlace")
+
+    return core.get_device_properties(device_id)
```

Comment:
Not friendly error message. If the users install the CPU version and he/she calls `paddle.device.cuda.get_device_properties()`, he/she would get the error message like `code has no attribute "get_device_properties"`. It is not friendly to users.